src/terminal.rs

```diff
@@ -0,0 +1,97 @@
+use std::io::{Stdin, Stdout, Write};
+use console::style;
+use console::Emoji;
+use crate::todo::Todo;
+
+
+pub enum TerminalError {
+  Stdin(std::io::Error),
+  Stdout(std::io::Error),
+}
+
+impl TerminalError {
+  pub fn show_error(&self) -> String {
+      match self {
+          TerminalError::Stdin(error) => format!("Erro de entrada: {}", style(error).red()),
+          TerminalError::Stdout(error) => format!("Erro de saída: {}",  style(error).red()),
+      }
+  }
+}
+pub struct Terminal {
+  stdin: Stdin,
+  stdout: Stdout,
+}
+
+impl Terminal {
+  pub fn new() -> Terminal {
+      Terminal {
+          stdin: std::io::stdin(),
+          stdout: std::io::stdout(),
+      }
+  }
+
+  pub fn run(&mut self) -> Result<(), TerminalError> {
+      writeln!(self.stdout, "{} {}", style("Bem vindo ao TODO List").yellow().bold(), Emoji("🌈", ""))
+          .map_err(|error| TerminalError::Stdout(error))?;
+
+      loop {
+          let new_todo = self.ask_for_new_todo()?;
+          if let Some(new_todo) = new_todo {
+              self.show_todo(&new_todo)?;
+              writeln!(self.stdout, "{} {}", style("TODO inserido com sucesso!").cyan(), Emoji("🥳", ""))
+                  .map_err(|error| TerminalError::Stdout(error))?;
+          } else {
+              break;
+          }
+      }
+      Ok(())
```

Comment:
_Muito_ nitpick, mas nesses casos, é melhor usar `return Ok(())` direto, ao invés de `break`. Deixa o código mais fácil de ler.

```suggestion
              return Ok(());
          }
      }
```